File: packages/hy_channel_info/ios/Classes/HyChannelInfoPlugin.m
#import "HyChannelInfoPlugin.h"

@implementation HyChannelInfoPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"hy_channel_info"
            binaryMessenger:[registrar messenger]];
  HyChannelInfoPlugin* instance = [[HyChannelInfoPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getChannelInfo" isEqualToString:call.method]) {
    result([[NSBundle mainBundle] objectForInfoDictionaryKey:@"channel_info"]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: packages/hy_channel_info/macos/Classes/HyChannelInfoPlugin.swift
import Cocoa
import FlutterMacOS

public class HyChannelInfoPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "hy_channel_info", binaryMessenger: registrar.messenger)
    let instance = HyChannelInfoPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getChannelInfo":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
